rclcpp/include/rclcpp/node.hpp

```diff
@@ -277,6 +277,12 @@ class Node : public std::enable_shared_from_this<Node>
     const std::string & name,
     const ParameterT & value);
 
+  template<typename MapValueT>
+  void
+  set_parameter_if_not_set(
```

Comment:
I don't feel strongly about the name, but I will point out that in ROS1 all of the methods were named the same, regardless of whether they were setting a map, a vector, or a single value.  I think there is value to this here as well, since there is only "one" name to remember, regardless of the type you want to set.  However, if you insist, I will change the name to be plural.